CMake/Platforms/WindowsCache.cmake

```diff
@@ -118,6 +118,11 @@ if(NOT UNIX)
 
     set(HAVE_SIGACTION 0)
     set(HAVE_MACRO_SIGSETJMP 0)
+
+    if(${CMAKE_CL_64})
+      add_definitions(D_AMD64 /Damd64)
```

Comment:
Is this really related to building statically? Is "having a 64 bit compiler from microsoft" really a enough to deduce that we're on an amd64 architecture? (which I assume the define would signify, there's no comment that explains what it's for)